lib/chef/knife/winrm_knife_base.rb

```diff
@@ -0,0 +1,316 @@
+              session_exit_code = session.exit_code
+              unless success_return_codes.include? session_exit_code.to_i
+                @exit_code = [@exit_code, session_exit_code.to_i].max
+                ui.error "Failed to execute command on #{session.host} return code #{session_exit_code}"
+              end
+            end
+          end
+
+          def success_return_codes
+            # Redundant if the CLI options parsing occurs
+            return [0] unless config[:returns]
+
+            @success_return_codes ||= config[:returns].split(",").collect(&:to_i)
+          end
+
+          def session_from_list
+            @list.each do |item|
+              Chef::Log.debug("Adding #{item}")
+              @session_opts[:host] = item
+              create_winrm_session(@session_opts)
+            end
+          end
+
+          def create_winrm_session(options = {})
+            session = Chef::Knife::WinrmSession.new(options)
+            @winrm_sessions ||= []
+            @winrm_sessions.push(session)
+          end
+
+          def resolve_session_options
+            config[:winrm_port] ||= ( config[:winrm_transport] == "ssl" ) ? "5986" : "5985"
+
+            @session_opts = {
+              user: resolve_winrm_user,
+              password: config[:winrm_password],
+              port: config[:winrm_port],
+              operation_timeout: resolve_winrm_session_timeout,
+              basic_auth_only: resolve_winrm_basic_auth,
+              disable_sspi: resolve_winrm_disable_sspi,
+              transport: resolve_winrm_transport,
+              no_ssl_peer_verification: resolve_no_ssl_peer_verification,
+              ssl_peer_fingerprint: resolve_ssl_peer_fingerprint,
+              shell: config[:winrm_shell],
+              codepage: config[:winrm_codepage],
+            }
+
+            if @session_opts[:user] && (not @session_opts[:password])
+              @session_opts[:password] = config[:winrm_password] = get_password
+            end
+
+            if @session_opts[:transport] == :kerberos
+              @session_opts.merge!(resolve_winrm_kerberos_options)
+            end
+
+            @session_opts[:ca_trust_path] = config[:ca_trust_file] if config[:ca_trust_file]
+          end
+
+          def resolve_winrm_user
+            user = config[:winrm_user]
+
+            # Prefixing with '.\' when using negotiate
+            # to auth user against local machine domain
+            if resolve_winrm_basic_auth ||
+                resolve_winrm_transport == :kerberos ||
+                user.include?("\\") ||
+                user.include?("@")
+              user
+            else
+              ".\\#{user}"
+            end
+          end
+
+          def resolve_winrm_session_timeout
+            # 30 min (Default) OperationTimeout for long bootstraps fix for KNIFE_WINDOWS-8
+            config[:session_timeout].to_i * 60 if config[:session_timeout]
+          end
```

Comment:
Can this coercion happen where the CLI option gets defined?

Reply:
Since we want folks to enter minutes and we need seconds not really. I renamed it though so it makes more sense